### ISSUE TITLE
Table editor changes for table column value not saved upon save 

### ISSUE THREAD

**Issue by symdeb** [dbeaver/dbeaver#19915]
### Description

Community 23.0.4

A table has no primary key. When changing a field of a record , the save button lights up. DBeaver uses all fields in the UPDATE command, however the data never gets updated upon a "save" even when the save button returns to grey color. This gives the use the impression that the change was saved. Quite a long time ago I could make changes with save and this problems did not happen. As a workaround now copy paste the update script and modify/generate the SQL. The save function makes changing much more easier. Also tried to change properties and enable "auto commit" but did not have effect.

P.S. Copying  the  generated script from the save button and copy/paste it into a new script and execute it, also resulted in 0 updates. Have not have time to analyze where the script is wrong.

### DBeaver Version

Community 23.0.4

### Operating System

Unbuntu 22.04

### Database and driver

 mysql  Ver 15.1 Distrib 10.8.7-MariaDB, for debian-linux-gnu (x86_64) using  EditLine wrapper

### Steps to reproduce

1. Click on table, real SQL in editor
2. change value of a column (Save higlhighted)
3. Click save dropdown and show the SQL
4. Click on Save
5. Save (save is dimmed)
6. Click on table, read SQL in edit
7. Values are the same as before the edit

**Comment by LonwoLonwo** [comment]
Hello @symdeb 

Unfortunately, I'm not able to reproduce your issue for now.

Could you please provide a screen from your settings? Preferences -> Editors -> the Data Editor page

Some footage of the issue also will be very helpful.

**Comment by E1izabeth** [comment]
It's been a while since no update here. Please, let me know if the issue is still actual and provide additional information.

**Comment by symdeb** [comment]
it seem to happen on tables that have columns that are float(10,10) . You may reopen if you  can reproduce it with this information ?

**Comment by E1izabeth** [comment]
I reproduced the issue. It's happening because float in the table doesn't match the float in update script.
![image](https://github.com/dbeaver/dbeaver/assets/28875055/38bf42d9-b96a-4359-bdb9-39a4900c7750)

**Comment by symdeb** [comment]
Great. I noticed that also before, Somehow when you enter a float, DBeaver (or MYSQL, MariaDB in my case) does not use the value but something nearby with a fractional difference .  I generated the update script from DBeaver itself. Does that mean this is an issue DBSeaver or in MySQL ? I believe i used the a setting as described here  https://github.com/dbeaver/dbeaver/issues/5077 but I recall even after that i it did not work, The real dangerous thing is tat DBeaver say Save as OK ,while it was not, and I only found it out after a lot of manual updates that ...... actually were not updated.
